omp/components/sort_small.hpp

```diff
@@ -46,29 +46,31 @@ namespace omp {
 
 
 #ifdef __GNUG__
-#define GKO_GCC_EXPECT_LIKELY(v) __builtin_expect(v, true)
+#define GKO_GNU_EXPECT_LIKELY(v) __builtin_expect(v, true)
 #else
-#define GKO_GCC_EXPECT_LIKELY(v) v
-#endif
+#define GKO_GNU_EXPECT_LIKELY(v) v
+#endif  // defined __GNUG__
 
 #ifdef _MSC_VER
-#define GKO_COMPARATOR(x, y)                            \
-    {                                                   \
-        const auto should_swap = !comp(arr[x], arr[y]); \
-        const auto tmp = arr[x];                        \
-        arr[x] = should_swap ? arr[y] : arr[x];         \
-        arr[y] = should_swap ? tmp : arr[y];            \
-    }
+template <typename TargetType, typename Functor>
+inline void comparator(TargetType *arr, size_type x, size_type y, Functor comp)
+{
+    const auto should_swap = !comp(arr[x], arr[y]);
+    const auto tmp = arr[x];
+    arr[x] = should_swap ? arr[y] : arr[x];
+    arr[y] = should_swap ? tmp : arr[y];
+}
```

Comment:
The only nit I have here is that this function should be in the namespace `detail` since it is a helper function.

Reply:
Just pushed - is the spacing correct?

Reply:
Yes, thank you!